File: src/bindings/commands/window.ts
import { invoke, isTauri } from '@tauri-apps/api/core'

/// Open the native settings window (desktop only).
export async function openSettingsWindow(): Promise<void> {
  await invoke<void>('open_settings_window')
}

/// Whether native window commands are available.
export function canOpenNativeWindow(): boolean {
  return isTauri()
}

/// Open (or focus) the main application window (desktop only).
export async function openAppWindow() {
  return await invoke<void>('open_app_window')
}

File: src-tauri/src/commands/window.rs
use crate::{error::AppError, window::WindowManager};

/// Open (or focus) the settings window.
#[tauri::command]
pub fn open_settings_window() -> Result<(), AppError> {
    WindowManager::global()
        .open_settings_window()
        .map_err(|e| AppError::window(e.to_string()))
}

/// Open (or focus) the main application window.
#[tauri::command]
pub fn open_app_window() -> Result<(), AppError> {
    WindowManager::global()
        .open_message_window()
        .map_err(|e| AppError::window(e.to_string()))
}

/// Open (or focus) the login window
#[tauri::command]
pub fn open_login_window() -> Result<(), AppError> {
    WindowManager::global()
        .open_login_window()
        .map_err(|e| AppError::window(e.to_string()))
}
